Otus-DevOps-2018-02/ifqthenp_microservices
# HW 22. Introduction to Kubernetes (kubernetes-1)

# Completed Homework №22


## The following tasks have been completed

- :large_blue_diamond: Created `yml` files in `kubernetes/reddit` folder with Deployment manifests for Reddit application:
  - `ui-deployment.yml`
  - `comment-deployment.yml`
  - `post-deployment.yml`
  - `mongo-deployment.yml`
- :large_blue_diamond: Successfully completed [Kubernetes The Hard Way](https://github.com/kelseyhightower/kubernetes-the-hard-way) tutorial from Kelsey Hightower
- :large_blue_diamond: Successfully tested Reddit application deployment to the Kubernetes cluster with `kubectl -f <filename>` command

## How to get project running

- Complete [Kubernetes The Hard Way](https://github.com/kelseyhightower/kubernetes-the-hard-way) tutorial 
- Run `kubectl apply -f <filename>.yml` from `kubernetes/reddit` folder

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Run `kubectl get pods`

## PR checklist
